rules/integrations/okta/credential_access_okta_mfa_bombing_via_push_notifications.toml

```
@@ -0,0 +1,88 @@
+
+#### Possible investigation steps:
+
+- Identify the user who received the MFA notifications by reviewing the `user.email` field.
+- Identify the time, source IP, and geographical location of the MFA requests and the subsequent successful login.
+- Review the `event.action` field to understand the nature of the events. It should include two `user.mfa.okta_verify.deny_push` actions and one `user.authentication.sso` action.
+- Ask the user if they remember receiving the MFA notifications and subsequently logging into their account.
+- Check if the MFA requests and the successful login occurred during the user's regular activity hours.
+- Look for any other suspicious activity on the account around the same time.
+- Identify whether the same pattern is repeated for other users in your organization. Multiple users receiving push notifications simultaneously might indicate a larger attack.
+
+### False positive analysis:
+
+- Determine if the MFA push notifications were legitimate. Sometimes, users accidentally trigger MFA requests or deny them unintentionally and later approve them.
+- Check if there are known issues with the MFA system causing false denials.
+
+### Response and remediation:
+
+- If unauthorized access is confirmed, initiate your incident response process.
+- Alert the user and your IT department immediately.
+- If possible, isolate the user's account until the issue is resolved.
+- Investigate the source of the unauthorized access.
+- If the account was accessed by an unauthorized party, determine the actions they took after logging in.
+- Consider enhancing your MFA policy to prevent such incidents in the future.
+- Encourage users to report any unexpected MFA notifications immediately.
+- Review and update your incident response plans and security policies based on the findings from the incident.
+
+## Setup
+
+The Okta Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule."""
+references = [
+    "https://www.mandiant.com/resources/russian-targeting-gov-business",
+    "https://www.elastic.co/security-labs/testing-okta-visibility-and-detection-dorothy",
+    "https://sec.okta.com/articles/2023/08/cross-tenant-impersonation-prevention-and-detection",
+    "https://www.rezonate.io/blog/okta-logs-decoded-unveiling-identity-threats-through-threat-hunting/",
+]
+risk_score = 73
+rule_id = "8a0fbd26-867f-11ee-947c-f661ea17fbcd"
+severity = "high"
+tags = ["Use Case: Identity and Access Audit", "Tactic: Credential Access", "Data Source: Okta"]
+type = "eql"
+
+query = '''
+sequence by okta.actor.id with maxspan=10m
+  [any where event.dataset == "okta.system" and event.module == "okta"
+    and event.action == "user.mfa.okta_verify.deny_push"] with runs=5
```

you need to ensure that there is no interim successful login within these, otherwise you could end up with FP situations where successful logins occurred with legitimate failures.

```suggestion
    and event.action == "user.mfa.okta_verify.deny_push"] with runs=5
    until [any where event.dataset == "okta.system" and event.module == "okta"
      and event.action == UPDATE-WITH-SUCCESS]
```

---

hmmm, ideally we want to catch MFA bombing even if it is not successful as it could indicate stolen credentials. Curious if negation as the last sequence for a successful login event would suffice to avoid the potential false positive scenario you mentioned.

---

What I am saying is that _with_ a successful login, it is less likely a true bombing attempt and may more likely be a legitimate failure (mistype, etc.).

The short circuit from `until` is better here than a negation, because with negation, the event would be position dependent

---

Added the following which should check for the successful authentication either via okta login or via an SSO application or a user session start.

```
  until [authentication where event.dataset == "okta.system"
    and ((okta.event_type == "user.authentication.sso" and okta.outcome.result == "SUCCESS")
      or (okta.event_type == "user.authentication.verify" and okta.outcome.result == "SUCCESS")
      or okta.event_type == "user.session.start")]
```

Thanks for the suggestion!